flang/lib/Lower/Bridge.cpp

```diff
@@ -1854,13 +1918,8 @@ class FirConverter : public Fortran::lower::AbstractConverter {
                   (rhsType && rhsType->IsPolymorphic()))
                 TODO(loc, "pointer assignment involving polymorphic entity");
 
-              if (explicitIterationSpace()) {
-                Fortran::lower::createAnyArrayPointerAssignment(
-                    *this, assign.lhs, assign.rhs, lbExprs, explicitIterSpace,
-                    implicitIterSpace, localSymbols);
-                return;
-              }
-
+              // FIXME: in the explicit space context, we want to use
+              // ScalarArrayExprLowering here.
```

Comment:
Does this mean that the FIR in the pointer assignment tests in forall-2.f90 is actually wrong now ?

Reply:
I don't think we're properly checking the conditions for creating copies. Trivial cases should be ok.